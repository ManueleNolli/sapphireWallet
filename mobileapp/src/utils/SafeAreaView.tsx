import React from 'react'
import { SafeAreaView as RNSafeAreaView } from 'react-native-safe-area-context'
import { vh, vw } from '../Styles'
import { Layout } from '@ui-kitten/components'

type SafeAreaViewProps = {
  children: React.ReactNode
  style?: any
}

export default function SafeAreaView({ children, style }: SafeAreaViewProps) {
  return (
    <Layout style={{ flex: 1 }}>
      <RNSafeAreaView
        style={[
          {
            flex: 1,
            marginVertical: 5 * vh,
            marginHorizontal: 5 * vw,
          },
          style,
        ]}
      >
        {children}
      </RNSafeAreaView>
    </Layout>
  )
}
