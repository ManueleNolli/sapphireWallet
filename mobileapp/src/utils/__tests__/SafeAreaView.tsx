import React from 'react'
import renderWithTheme from '../../TestHelper'
import { Text } from 'react-native'
import SafeAreaView from '../SafeAreaView'

describe('SafeAreaView', () => {
  it('Render correctly', () => {
    const tree = renderWithTheme(
      <SafeAreaView>
        <Text>Test</Text>
      </SafeAreaView>
    )
    expect(tree).toMatchSnapshot()
  })
})
