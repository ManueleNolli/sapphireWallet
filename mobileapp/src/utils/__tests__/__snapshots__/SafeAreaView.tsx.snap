// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeAreaView Render correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "flex": 1,
          "marginHorizontal": 37.5,
          "marginVertical": 66.7,
        },
        undefined,
      ]
    }
  >
    <Text>
      Test
    </Text>
  </RNCSafeAreaView>
</View>
`;
